### PR TITLE
Issue #3 | Creates views  "add" & " list", and routes between them

### LINKED ISSUE #3: 2. As a user, I want to be able to navigate between the “list” view and the “add an item” view so that the interface is clearly delinteated between viewing and adding.
This is 1 of 2 stories that should be completed in the first week to enable further development of the app.

AC:

* `react-router-dom` has been added as a project dependency
* Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
* When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed
* Whichever view is selected, the corresponding link should display in bold text

### PR DESCRIPTION
**Pull request by apdsrocha**
## Description

**What does this code change:**
This adds two different views and routes so that the user is able to switch views (rendering different components) through clicking the corresponding links. Aside from being able to render these different components through their own routes, the styling of the links changes according to the view being rendered. To achieve the styling change, our first approach was to implement a CSS class toggle while tracking state, but after checking out the `react router` documentation we made use of the `<NavLink>` tag and achieved the same result implementing a much cleaner solution. It was a great learning opportunity to try something new and make good use of the documentation.

## Related Issue

Closes #3 

## Acceptance Criteria

- [x]  `react-router-dom` has been added as a project dependency
- [x] Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
- [x] When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed
- [x] Whichever view is selected, the corresponding link should display in bold text

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

00_UI_before

### After

01_UI_after
02_UI_after

## Testing Steps / QA Criteria

* Switch from the main branch to the `jc-ar-create-routes` branch
* Pull the changes from this branch
* Run `npm update` to make sure the current dependencies are installed
* Run `npm start` to run the local host
* Go to localhost:3000 in your browser

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,26 @@
...
+        <Switch>
+          <Route exact path="/">
+            <List />
+          </Route>
+
+          <Route exact path="/add-item">
+            <AddItem />
+          </Route>
+        </Switch>
+        <NavBar />
```
Comment (yenly) [inline src/App.js:21]: Good job in making the nav elements as their own component.

File: src/components/NavBar.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+      <RouterLink
+        className="nav-link"
+        exact
+        activeClassName="nav-link-bold"
+        to="/add-item"
+      >
+        Add item
+      </RouterLink>
+    </nav>
```
Comment (yenly) [inline src/components/NavBar.js:23]: Good job on using HTML `nav` element.

**Review by yenly — APPROVED** [review]
Great concise write-up! Looks like you set up a good foundation for the components to build from.

**Comment by jssckbl** [comment]
I wanted to at least comment on the pull request, even though it has long been approved! 

This looks great and everything works as expected. I also like that you two chose to set up the links with a navbar setup.

The only thing I wanted to ask about or suggest is the Route exact path for `<List />` in `App.js` for example:

```
          <Route exact path="/">
            <List />
          </Route>
```

Since `List.js` exists as a separate file, would it make sense to have the exact path be `"/list"` since an exact path of `"/"` typically would mean some sort of a home or start location on a website/app?

**Pull request merged**